diff --git a/src/server/ns_turn_server.c b/src/server/ns_turn_server.c
index 7b90bdc..1dbd4b4 100644
--- a/src/server/ns_turn_server.c
+++ b/src/server/ns_turn_server.c
@@ -1807,8 +1807,11 @@
 			ioa_network_buffer_set_size(nbh, len);
 		}
 
-		/* We add integrity for both long-term and short-term indication messages */
-		/* if(server->ct == TURN_CREDENTIALS_SHORT_TERM) */
+		/*
+		 * Don't add MESSAGE-INTEGRITY attribute when long term credentials are being used
+		 * as it breaks some turn clients.
+		 */
+		if(server->ct == TURN_CREDENTIALS_SHORT_TERM)
 		{
 			adjust_shatype(server,ss);
 			stun_attr_add_integrity_str(server->ct,ioa_network_buffer_data(nbh),&len,ss->hmackey,ss->pwd,ss->shatype);
@@ -4095,8 +4098,11 @@
 					ioa_network_buffer_set_size(nbh, len);
 				}
 
-				/* We add integrity for both long-term and short-term indication messages */
-				/* if(server->ct == TURN_CREDENTIALS_SHORT_TERM) */
+				/*
+				 * Don't add MESSAGE-INTEGRITY attribute when long term credentials are being used
+				 * as it breaks some turn clients.
+				 */
+				if(server->ct == TURN_CREDENTIALS_SHORT_TERM)
 				{
 					adjust_shatype(server,ss);
 					stun_attr_add_integrity_str(server->ct,ioa_network_buffer_data(nbh),&len,ss->hmackey,ss->pwd,ss->shatype);
